### PR TITLE
Introduce Schemas to handle missing body, empty transaction list, missing fields, additional date formats

### PR DESCRIPTION
**Pull request by MattFaz**
Added Schema functionality to allow easier validating of requests and responses.
No body, missing transactions, or missing fields will now return appropriate error messages _(see screenshots attached)_.

image
image
image

### CODE REVIEW

**Comment by MattFaz** [comment]
Transactions can now handle multiple date formats:
- `MMM DD, YYYY`
- `MMM DD YYYY`
- `YYYY-MM-DD` (ISO)

image

File: core/security.py [hunk 1]
```diff
@@ -20,9 +16,3 @@ async def get_api_key(api_key_header: str = Security(api_key_header)):
             status_code=403,
             detail="Could not validate credentials",
         )
-
-
-def generate_custom_id():
```
Comment (MattFaz) [inline core/security.py:25]: Removed as its only used in 1 function, and we can use the default python `uuid` library

File: api/transactions.py [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
 
-from core.security import get_api_key
-from models.transaction import Transaction
+from schemas.transactions import Transaction
 from services.actual_service import actual_service
 
 router = APIRouter()
 
 
-@router.post("/transactions", dependencies=[Depends(get_api_key)])
```
Comment (MattFaz) [inline api/transactions.py:16]: Removed as `/transaction` and `/transactions/` will work regardless, no need to add both.
Additionally we do not need the dependency on `get_api_key` here as it's already added in `main.py` when we add the route

**Comment by MattFaz** [comment]
@bobokun happy to discuss anything here :)

**Review by bobokun — APPROVED** [review]
Thanks for the welcomed changes! Sorry for the delay in reviewing this but it looks good to me

File: api/transactions.py (hunk 2 above)
Comment (bobokun) [reply api/transactions.py:16]: Hmm, I realized this doesn't work actually when testing with Ios shortcuts. Yes `/transactions` and `/transactions/` work in postman but after merging this PR it seems to have broken the original shared IOS shortcut which uses `/transactions/` I also realized you removed `dependencies=[Depends(get_api_key)])` which we would want to keep in order to protect the endpoint with the api.

**Pull request merged**